### PR TITLE
DRIVERS-1539 Cleanup open server transactions in unified unpinning tests

### PR DESCRIPTION
**Pull request by ShaneHarvey**

### CODE REVIEW

**Review by comandeo — APPROVED** [review]
👍 It fixes the problem with open transactions.

File: source/transactions/tests/unified/mongos-unpin.json [hunk 1]
```diff
@@ -142,7 +160,7 @@
       ]
     },
     {
-      "description": "unpin after TransientTransactionError error on abort",
+      "description": "unpin after non-transient error on abort",
```
Comment (iwysiu) [inline source/transactions/tests/unified/mongos-unpin.json:163]: I assume these were mis-labeled?

**Review by iwysiu — APPROVED** [review]
lgtm!

File: source/transactions/tests/unified/mongos-unpin.json (hunk 1 above)
Comment (ShaneHarvey) [reply source/transactions/tests/unified/mongos-unpin.json:163]: Yep. This test uses the non-transient code 24 error.

**Comment by ShaneHarvey** [comment]
Waiting for this patch build to pass before merging: https://spruce.mongodb.com/version/60d6175757e85a312d907f61/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Pull request merged**